tests/test_content_manifest.py

```diff
@@ -245,6 +242,47 @@ def test_process_pip(default_request, default_toplevel_purl):
     assert cm._pip_data == expected_contents
 
 
+def test_process_rubygems(default_request, default_toplevel_purl):
+    packages_json = [
+        {
+            "name": "zync",
+            "type": "rubygems",
+            "version": "2.24.0",
+            "dependencies": [
+                {
+                    "name": "httpclient",
+                    "type": "rubygems",
+                    "version": f"git+https://github.com/3scale/httpclient.git@{DEP_COMMIT_ID}",
+                    "path": "some/path"
+                },
+                {"name": "zeitwerk", "type": "rubygems", "version": "2.4.2"},
+            ],
+        },
+    ]
+
+    packages = _load_packages_from_json(packages_json)
+    package = packages[0]
+
+    expected_purl = default_toplevel_purl
+    expected_git_purl = f"pkg:github/3scale/httpclient@{DEP_COMMIT_ID}"
+    expected_gem_purl = "pkg:gem/zeitwerk@2.4.2"
+
+    cm = ContentManifest(default_request, packages)
+    cm.to_json()
+
+    expected_contents = {
+        package: {
+            "purl": expected_purl,
+            "dependencies": [{"purl": expected_gem_purl}, {"purl": expected_git_purl}],
+            "sources": [{"purl": expected_gem_purl}, {"purl": expected_git_purl}],
+        },
+    }
```

Comment:
@chmeliik As I was implementing pushing of dependencies to temporary repo, I realized, it would be a good idea to think about the way returned metadata will be used for content manifest generation. What do you think I should do about PATH dependencies (dependency that is part of the top level repo itself, its source code is in some of the subdirectories of the project)? Should they be part of the generated content manifest? If so, should they be part of both "dependencies" and "sources" and do you have any preference on how to create a purl for this type, since there is nothing in the PURL spec about them?

Reply:
PURLs do support relative paths from the package root, so you can do something like `<parent_purl>#some/path`.

Here's how Cachito does it for gomod:
https://github.com/containerbuildsystem/cachito/blob/fe5633ad782b407d99c9243270dd42bc7d91fa85/cachito/web/content_manifest.py#L322-L325
https://github.com/containerbuildsystem/cachito/blob/fe5633ad782b407d99c9243270dd42bc7d91fa85/cachito/web/content_manifest.py#L72

They should be in both "dependencies" and "sources", unless they are non-production dependencies (dev / test etc.) in which case they would go only in "sources" (but AFAIK we won't be detecting dev vs. non-dev for Ruby, so they should go in both)

Reply:
Note that the relative paths must not contain `..` though, I don't remember if we allow PATH deps from parent directories or not but if we do, the PARENT_PURL would probably have to be a purl of the repo itself (not the package which is in some subpath of the repo)

(see https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#rules-for-each-purl-component)

Reply:
Thank you very much